yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,269 @@
+        pass
+
+
+class CFRadialDataset(Dataset):
+    _index_class = CFRadialHierarchy
+    _field_info_class = CFRadialFieldInfo
+
+    def __init__(
+        self,
+        filename,
+        dataset_type="cf_radial",
+        grid_shape=(40, 200, 200),
+        grid_limits=((0.0, 2000.0), (-1e5, 1e5), (-1e5, 1e5)),
+        storage_filename=None,
+        units_override=None,
+    ):
+        self.fluid_types += ("cf_radial",)
+        self._handle = xr.open_dataset(filename)
+        self.refine_by = 2
+        new_filename = filename[:-3] + "_grid.nc"
+        if "x" not in self._handle.coords:
+            if not os.path.isfile(new_filename):
+                from yt.utilities.on_demand_imports import _pyart
+
+                pyart = _pyart.pyart
+
+                radar = pyart.io.read_cfradial(filename)
+                self.grid_shape = grid_shape
+                self.grid_limits = grid_limits
+                grid = pyart.map.grid_from_radars(
+                    (radar,), grid_shape=self.grid_shape, grid_limits=self.grid_limits
+                )
+                mylog.warning(
+                    "Saving a cartesian grid for file %s at %s. Data will be loaded "
+                    "from the cartesian grid.",
+                    filename,
+                    new_filename,
+                )
+                grid.write(new_filename)
+            self._handle = xr.open_dataset(new_filename)
+            filename = new_filename
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.storage_filename = storage_filename
+        # refinement factor between a grid and its subgrid
+        # self.refine_by = 2
+
+    def _set_code_unit_attributes(self):
+        # This is where quantities are created that represent the various
+        # on-disk units.  These are the currently available quantities which
+        # should be set, along with examples of how to set them to standard
+        # values.
+        #
+        # self.length_unit = self.quan(1.0, "cm")
+        #
+        # These can also be set:
+        # self.velocity_unit = self.quan(1.0, "cm/s")
+        # self.magnetic_unit = self.quan(1.0, "gauss")
+        length_unit = self._handle.variables["x"].attrs["units"]
+        self.length_unit = self.quan(1.0, length_unit)
+        self.mass_unit = self.quan(1.0, "kg")
+        self.time_unit = self.quan(1.0, "s")
+
+    def _parse_parameter_file(self):
+        # This needs to set up the following items.  Note that these are all
+        # assumed to be in code units; domain_left_edge and domain_right_edge
+        # will be converted to YTArray automatically at a later time.
+        # This includes the cosmological parameters.
+
+        #
+        #   self.unique_identifier      <= unique identifier for the dataset
+        #                                  being read (e.g., UUID or ST_CTIME)
+        self.unique_identifier = int(os.stat(self.parameter_filename)[stat.ST_CTIME])
```

Comment:
this was copy-pasted in a lot of frontends, and it's now redundant with the `unique_identifier` property defined in the base class 
```suggestion
```